### ISSUE TITLE
windows_exporter upgrade fails on rancher/windows_exporter-package:v0.0.1 image already present

### ISSUE THREAD

**Issue by kravciak** [rancher/rancher#31827]
**To reproduce:**

- Deploy a windows cluster - 3 linux nodes - 1 etcd, 1 control and 1 worker nodes  (I added 10 windows worker nodes, probably not necessary)
- Point the feature charts to dev-v2.5 branch

- Deploy rancher-wins-upgrader with the following override:
```
masquerade:
  enabled: true
  as: c:\\etc\wmi-exporter\wmi-exporter.exe
```
- Deploy Monitoring v2 on the cluster into System Project
- Upgrade rancher-wins-upgrader app from Apps & Marketplace -> Installed apps -> Edit / Upgrade menu

**Current result**
All rancher-monitoring-windows-exporter pods go into Crashloopbackoff

```
kubectl describe pod -n cattle-monitoring-system rancher-monitoring-windows-exporter-25p6m
  Warning  BackOff  58m                    kubelet  Back-off restarting failed container
  Normal   Pulled   56m (x4 over 57m)      kubelet  Container image "rancher/windows_exporter-package:v0.0.1" already present on machine
```

**Environment information**
- Rancher version v2.5.8-rc1 and 2.5.7
- Windows: ami-008ec03d9035c8ee7
- Installation option single install (docker)

**Edit by thehejik 31/03/2021 below:** 
(just tried the upgrade way instead of reinstall and it leads to the same result):
**Result:**
* Once the `wins-upgrader` is redeployed/upgraded on all windows nodes with disabled masquerade observe status of pods in `rancher-monitoring-windows-exporter` DaemonSet - all pods are crashing due following error in `exporter-node` container:
```
FATA[2021-03-31T12:19:42Z] rpc error: code = NotFound desc = could not found binary: could not get checksum: exec: "c:\\etc\\windows-exporter\\windows-exporter.exe": file does not exist
```
**Other details that may be helpful:**
Workaround is to manually delete all the pods in `rancher-monitoring-windows-exporter` DaemonSet.

**Comment by aiyengar2** [comment]
Was able to replicate this issue. This was a tricky bug, but I suspect the reason why this happens is because wins [moves the binary from its original location on the host to another location for the process](https://github.com/rancher/wins/blob/b1e2d886d3e86642e3abfe40b2d127129325ceb8/pkg/apis/process_service.go#L38); therefore, the copy-binary step needs to happen once again in order if the container is restarted or else the binary that we point to won't exist on the host.

@kravciak can you also check the following scenario to see if it impacts Monitoring V1? Instead of using masquerade (which would block the wmi_exporter.exe path), just spin up a cluster in 2.5.8, deploy wins-upgrader without masquerading, enable Monitoring V1, and upgrade wins-upgrader with some sort of dummy update (e.g. add a random path to the whitelist prefix).

I suspect that the same issue will arise, in which case we might need to propagate some changes to all Windows supported charts to change the copy-binary step from an initContainer to a regular container.

cc: @luthermonson @nickgerace

**Comment by kravciak** [comment]
Hi, I am trying to deploy monitoring v1 from dev-2.5, but it's stack on 
```
ImagePullBackOff: Back-off pulling image "rancher/library-nginx:1.19.9-alpine"
```

**Comment by kravciak** [comment]
After changing pod image to library-nginx:1.19.2-alpine monitoring started fine.

Then I started upgrade of wins-upgrader (without any change in yaml), result is simillar:

All exporter-node-windows-cluster-monitoring pods are CrashLoopBackOff, logs contain:
```
FATA[2021-04-09T12:57:38Z] rpc error: code = NotFound desc = could not found binary: could not get checksum:
    exec: "c:\\etc\\wmi-exporter\\wmi-exporter.exe": file does not exist
```

**Comment by kravciak** [comment]
I tested the fix on 2.5.8-rc3 and rancher-monitoring-windows-exporter pods are fine after rancher-wins-upgrader upgrade
